dcore/diagnostics.py

```diff
@@ -0,0 +1,19 @@
+from firedrake import assemble, dot, dx
+from math import sqrt
+
+
+class Diagnostics(object):
+
+    def __init__(self, fieldlist=None):
```

Comment:
You could do:

```python
def __init__(self, *fields):
    self.fields = list(fields)

def register(self, *fields):
    self.fields += fields
```

That way the caller doesn't need to wrap fields up in a list.